uds/uds_communications/Uds/Uds.py

```diff
@@ -132,13 +133,19 @@ def send(self, msg, responseRequired=True, functionalReq=False, tpWaitTime=0.01)
 
         # Note: in automated mode (unlikely to be used any other way), there is no response from tester present, so threading is not an issue here.
         response = None
-        
+        self.last_resp_time = None
+        self.last_pending_resp_times = []
+
         if responseRequired:
             while True:
                 response = self.tp.recv(self.__P2_CAN_Client)
+                current_time = time.perf_counter() - before_send_time
+                if response[2] == 0x78:
+                    self.last_pending_resp_times.append(current_time)
                 if not ((response[0] == 0x7F) and (response[2] == 0x78)):
+                    self.last_resp_time = current_time
```

Comment:
Wasn't the idea here to progressively move the UDS-specific code from kiso-testing to this one?

We could directly return a UdsResponse here that contains the received response pending and positive response timestamps and that also includes a nice repr so that we don't have to work around the logging in pykiso